assessment/report_generator.go

```diff
@@ -189,3 +196,106 @@ func sourceColumnDefinitionToString(columnDefinition utils.SrcColumnDetails) str
 	}
 	return s
 }
+
+// TODO move calculation logic to assessment engine
+func calculateTableDbChangesAndImpact(srcTable utils.TableDetails, spTable utils.TableDetails) (string, string) {
+	changes := []string{}
+	impact := []string{}
+	if !strings.Contains(srcTable.Charset, "utf8") { // TODO add charset level comparisons - per source
+		changes = append(changes, "charset")
+		impact = append(impact, "storage increase")
+	}
+	if len(changes) == 0 {
+		changes = append(changes, "None")
+	}
+	return strings.Join(changes, ","), strings.Join(impact, ",")
+}
+
+// TODO move calculation logic to assessment engine
+func calculateColumnDbChangesAndImpact(srcCol utils.SrcColumnDetails, spCol utils.SpColumnDetails) (string, string) {
+	changes := []string{}
+	impact := []string{}
+	if srcCol.Datatype != spCol.Datatype { // TODO type specific checks on size
+		changes = append(changes, "type")
+
+		if getSrcColSizeBytesMySQL(srcCol) < getSpColSizeBytes(spCol) {
+			impact = append(impact, "storage increase")
+		}
+	}
+
+	//TODO Add check for unsigned
+	//TODO add check for not null to null scenarios
+	//TODO add check for size overflow
+	//TODO add diffs in modifiers and features - generated cols, auto inc, default etc
+
+	if len(changes) == 0 {
+		changes = append(changes, "None")
+	}
+	if len(impact) == 0 {
+		impact = append(impact, "None")
+	}
+	return strings.Join(changes, ","), strings.Join(impact, ",")
+}
+
+// TODO - move to source specific interfaces - also account for charsets
+func getSrcColSizeBytesMySQL(srcCol utils.SrcColumnDetails) int64 {
+	switch strings.ToLower(srcCol.Datatype) {
+	case "date":
+		return 4
+	case "timestamp":
+		return 4
+	case "bit":
+		return srcCol.Mods[0] / 8
+	case "int":
+		return 4
+	case "integer":
+		return 4
+	case "float":
+		return 4 // Add precision pspecific handling
+	case "text":
+		return 2 ^ 16
+	case "mediumtext":
+		return 2 ^ 24
+	case "longtext":
+		return 2 ^ 32
+	default:
+		//TODO - add all types
+		return 4
+	}
+}
+
+// TODO - move to source specific interfaces - also account for charsets
+func getSpColSizeBytes(spCol utils.SpColumnDetails) int64 {
+	var size int64
+	switch strings.ToUpper(spCol.Datatype) {
+	case "ARRAY":
+		size = spCol.Len
```

Comment:
In case of array datatype , datatype would be string , int, etc and IsArray would be set as true

Reply:
the current flow works if 1 byte records are used. this logic will be a lot more involved based on underlying type. added a todo here to take up in a follow up PR